custom_components/weback_vacuum/vacdevice.py

```diff
@@ -58,9 +59,19 @@ async def watch_state(self):
                 "Error on watch_state starting refresh_handler %s", watch_excpt
             )
 
+    def on_message(self, ws, message):
+        # Call the function on webackapi first, it'll update the status.
+        super().on_message(ws, message)
+
+        # Some vacuums won't have hismap_id in the initial status, but will
+        # report it later on. Let's make sure we try to load map again then.
+        if self.ACTIVE_MAP_ID_PROP in self.robot_status and not self.map:
+            asyncio.run(self.load_maps())
```

Comment:
I'm not 100% sure it's a good idea, but this mix of async and sync functions drives me nuts.